### PR TITLE
fix resource leak when load plugin failed

### PR DESCRIPTION
**Pull request by blueblueblue9**

### CODE REVIEW

**Review by jforissier — COMMENTED** [review]
Hello @blueblueblue9, thanks for your patch. Could you please add a `tee-supplicant:` prefix to the commit subject, as well as some description and your Signed-off-by: tags? Then you can add my:
`Reviewed-by: Jerome Forissier <jerome.forissier@linaro.org>`

For example:
```
tee-supplicant: fix resource leak when load plugin failed

load_plugin() calls dlopen() but does not call dlclose() when dlsym()
fails. Fix that.

Signed-off-by: Liu Yi <liu_yi@kylinos.cn>
Reviewed-by: Jerome Forissier <jerome.forissier@linaro.org>
```

**Comment by blueblueblue9** [comment]
update commit logs and Signed-off-by: tags

File: tee-supplicant/src/plugin.c [hunk 1]
```diff
@@ -78,14 +78,15 @@ static enum plugin_err load_plugin(const char *name, struct plugin *p)
 
 	handle = dlopen(name, RTLD_LAZY);
 	if (!handle)
-		return PLUGIN_DL_OPEN_ERR;
-
-	p->handle = handle;
+		return PLUGIN_DL_OPEN_ERR;
```
Comment (etienne-lms) [inline tee-supplicant/src/plugin.c:81]: remove trailing tabulation

**Review by etienne-lms — COMMENTED** [review]
`Reviewed-by: Etienne Carriere <etienne.carriere@linaro.org>`
In commit message: s/wh`en/when/.
Also:
```patch
-- load_plugin() calls dlopen() but does not call dlclose() wh`en dlsym()
+load_plugin() calls dlopen() but does not call dlclose() when dlsym()
 fails. Fix that.

**Comment by jforissier** [comment]
Formatting issues fixed and commit merged. Thanks @blueblueblue9.

**Pull request closed without merging**